cmake/modules/AddSwift.cmake

```diff
@@ -133,7 +133,9 @@ function(_add_host_variant_c_compile_flags target)
   is_build_type_with_debuginfo("${CMAKE_BUILD_TYPE}" debuginfo)
 
   # Add -O0/-O2/-O3/-Os/-g/-momit-leaf-frame-pointer/... based on CMAKE_BUILD_TYPE.
-  target_compile_options(${target} PRIVATE "${CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}}")
+  string(TOUPPER ${CMAKE_BUILD_TYPE} build_type_upper)
+  string(REPLACE " " ";" cxx_flags_list "${CMAKE_CXX_FLAGS_${build_type_upper}}")
```

Comment:
I believe that you can pass the list without the conversion here.

Reply:
Nope, had that before, failed terribly: <https://ci.swift.org/job/swift-PR-Linux/23670/console>

Reply:
Ah, unfortunate.